storage/src/test/java/tech/pegasys/teku/storage/server/kvstore/dataaccess/VersionedHashDBSourceTest.java

```diff
@@ -0,0 +1,351 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.storage.server.kvstore.dataaccess;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.IntStream;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.tuweni.bytes.Bytes;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.kzg.KZGCommitment;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.TestSpecFactory;
+import tech.pegasys.teku.spec.datastructures.blobs.DataColumnSidecar;
+import tech.pegasys.teku.spec.datastructures.blobs.versions.deneb.BlobSidecar;
+import tech.pegasys.teku.spec.datastructures.blobs.versions.fulu.DataColumnSidecarFulu;
+import tech.pegasys.teku.spec.datastructures.blocks.SlotAndBlockRoot;
+import tech.pegasys.teku.spec.datastructures.util.DataColumnSlotAndIdentifier;
+import tech.pegasys.teku.spec.datastructures.util.SlotAndBlockRootAndBlobIndex;
+import tech.pegasys.teku.spec.logic.versions.deneb.types.VersionedHash;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+import tech.pegasys.teku.storage.api.SidecarIdentifier;
+
+@SuppressWarnings("unchecked")
+public class VersionedHashDBSourceTest {
+  private final Spec spec = TestSpecFactory.createMinimalFulu();
+  private final DataStructureUtil dataStructureUtil = new DataStructureUtil(spec);
+  private final KvStoreCombinedDao dao = mock(KvStoreCombinedDao.class);
+  private final Function<KZGCommitment, VersionedHash> commitmentToVersionedHashFunction =
+      mock(Function.class);
+  private final VersionedHash versionedHash = dataStructureUtil.randomVersionedHash();
+  private VersionedHashDBSource versionedHashDBSource =
+      new VersionedHashDBSource(dao, commitmentToVersionedHashFunction, spec);
+
+  @BeforeEach
+  public void setup() {
+    when(commitmentToVersionedHashFunction.apply(any())).thenReturn(versionedHash);
+    versionedHashDBSource = new VersionedHashDBSource(dao, commitmentToVersionedHashFunction, spec);
+  }
+
+  @Test
+  public void storeSidecarHashes_isFunctional() {
```

Comment:
isFunctional is kinda redundant on all of these. i'd be happy with the first part i think